### PR TITLE
[Test refactor 4/5] Improve the scheduled tests

### PR DESCRIPTION
**Pull request by LysandreJik**
The scheduled tests now leverage a docker container pre-built two hours earlier than this run by the CI.

It generates a job for each folder in the `tests` folder.

### CODE REVIEW

File: .github/workflows/self-scheduled.yml [hunk 1]
```diff
@@ -3,533 +3,256 @@ name: Self-hosted runner (scheduled)
...
+        continue-on-error: true
+        run: cat /transformers/reports/${{ matrix.machines }}_tests_gpu_${{ matrix.folders }}/failures_short.txt
 
       - name: Test suite reports artifacts
         if: ${{ always() }}
         uses: actions/upload-artifact@v2
         with:
-          name: run_all_tests_tf_gpu_test_reports
-          path: reports
+          name: ${{ matrix.machines }}_run_all_tests_gpu_${{ matrix.folders }}_test_reports
```
Comment (patrickvonplaten) [inline .github/workflows/self-scheduled.yml:104]: Nice!

**Comment by patrickvonplaten** [comment]
Looks good to me! Also great that all `generation` tests from `test_generation_utils.py` are run! The errors:

```
ImportError: cannot import name 'json' from 'itsdangerous' (/usr/local/lib/python3.8/dist-packages/itsdangerous/__init__.py)
```

will be fixed once the previous PRs are merged?

**Comment by patrickvonplaten** [comment]
> Looks good to me! Also great that all `generation` tests from `test_generation_utils.py` are run! The errors:
> 
> ```
> ImportError: cannot import name 'json' from 'itsdangerous' (/usr/local/lib/python3.8/dist-packages/itsdangerous/__init__.py)
> ```
> 
> will be fixed once the previous PRs are merged?

Actually encountered this problem locally now as well -> think we need to pin the `itsdangerous` library: https://serverfault.com/questions/1094062/error-from-itsdangerous-import-json-as-json-importerror-cannot-import-name-j

**Review by sgugger — APPROVED** [review]
Thanks for working on this!

**Review by sgugger — APPROVED** [review]

**Review by patrickvonplaten — APPROVED** [review]

**Pull request merged**